File: ctci_1_3.swift
#!/usr/bin/env swift
// cracking the coding interview q 1.3
// determine if a string is a permutation of another

func isPermutation(x: String, y: String) -> Bool {
  let charactersX = Array(x).sorted { $0 < $1 }
  let charactersY = Array(y).sorted { $0 < $1 }
  return charactersX == charactersY
}

println(isPermutation("asdf","fdsa"))
println(isPermutation("hello","world"))

File: ctci_1_3.js
#!/usr/bin/env node

function isPermutation(one,two) {
  var oneChars = one.split("").sort().map(function(c) { return c.toLowerCase() });
  var twoChars = two.split("").sort().map(function(c) { return c.toLowerCase() });
  if (oneChars.length !== twoChars.length) { return false };
  for (var i in oneChars) {
    if (oneChars[i] !== twoChars[i]) { return false };
  }
  return true;
}

console.log(isPermutation("Hello","World"));
console.log(isPermutation("asdf","fsda"));
